src/net/sourceforge/kolmafia/CoinmasterData.java

```diff
@@ -6,119 +6,662 @@
-      final List<AdventureResult> buyItems,
-      final Map<Integer, Integer> buyPrices,
-      final String sellURL,
-      final String sellAction,
-      final List<AdventureResult> sellItems,
-      final Map<Integer, Integer> sellPrices,
-      final String itemField,
-      final Pattern itemPattern,
-      final String countField,
-      final Pattern countPattern,
-      final String storageAction,
-      final String tradeAllAction,
-      final boolean canPurchase) {
+      String master, final String nickname, final Class<? extends CoinMasterRequest> requestClass) {
     this.master = master;
     this.nickname = nickname;
     this.requestClass = requestClass;
+  }
+
+  // Fluid field construction
+
+  /**
+   * Defines the token used by the coinmaster,
+   *
+   * <p>A coinmaster deals in currencies other than Meat. If there is only one such currency, we
+   * call it the "token"
+   *
+   * @param token - Token used
+   * @return this - Allows fluid chaining of fields
+   */
+  public CoinmasterData withToken(String token) {
     this.token = token;
+    return this;
+  }
+
+  /**
+   * Defines the plural name of the coinmaster's token.
+   *
+   * <p>The plural name is ordinarily lazily derived by adding "s" to the end of the token's name,
+   * or, if it is an item, using the item's plural name. If that does not suffice, specifying it
+   * here overrides lazy derivation.
+   *
+   * @param pluralToken - Plural name for the token.
+   * @return this - Allows fluid chaining of fields
+   */
+  public CoinmasterData withPluralToken(String pluralToken) {
+    this.pluralToken = pluralToken;
+    return this;
+  }
+
+  /**
+   * Defines text that indicates you have no tokens available to spend.
+   *
+   * <p>When examining the responseText that shows the coinmaster's inventory, usually the available
+   * amount of each currency is displayed.
+   *
+   * <p>If there is only one currency - the token - and there is a simple string indicating that
+   * none are available, provide it here. If it is present, we need not do a pattern search to
+   * calculate the number of available tokens.
+   *
+   * <p>This is mutually exclusive with <code>positive</code>. Provide one or the other (or
+   * neither).
+   *
+   * @param tokenTest - String indicating zero tokens available
+   * @return this - Allows fluid chaining of fields
+   */
+  public CoinmasterData withTokenTest(String tokenTest) {
     this.tokenTest = tokenTest;
+    return this;
+  }
+
+  /**
+   * Defines text that indicates you have at least one token available.
+   *
+   * <p>Occasionally (we have one example), a coinmaster has a single token, but instead of a simple
+   * string indicating no tokens are available, there is a string indicating you have at least one.
+   *
+   * <p>This is mutually exclusive with <code>tokenTest</code>. Provide one or the other (or
+   * neither).
+   *
+   * @param positiveTest - String indicating at least one token is available
```

Comment:
it's not a string, but a boolean indicating whether "tokenTest" is a positive or negative test

Reply:
Oh, right.